Adding another option for GET endpoints

DanMMX opened the pull request:
For my apps I'd like to have the GET methods return a `{ count, rows }` object so I can create paginations easily.

Also, removed lodash from the generated project since we can use the `Object.assign` method. where lodash was used

diegohaz commented on generators/api/templates/index.js:
```diff
@@ -73,7 +73,12 @@ const { <%= modelFields.join(', ') %> } = schema.tree
+ <%_ } else { _%>
+ * @apiSuccess {Number} count Total amount of <%= lowers %>.
+ * @apiSuccess {Object[]} rows List of <%= lowers %>.
+ <%_ } _%>
```
Would not it be the other way around?

DanMMX replied:
You're right! fixed (forced push)

diegohaz commented:
Amazing job you've done here, @DanMMX.

I'm not sure if we can get rid of `_.merge` though. That's not exactly the same as `Object.assign` (which would be `_.assign`).

I didn't test it yet, but I would guess that `config.js` is broken.

https://lodash.com/docs/4.17.4#merge

DanMMX commented:
Even though `Object.assign` is not exactly the same as `_.merge`, the intention of the places where I replaced it gives us the needed behavior:

- [generators/api/templates/controller.js](https://github.com/diegohaz/rest/pull/78/files#diff-1b2a16fa314ee7916b33023c3bc7361fR79)
- [generators/app/templates/api/user/controller.js](https://github.com/diegohaz/rest/pull/78/files#diff-0e3d5804ce0dfc66969d0fdbdabbaae0R65)
- [generators/app/templates/src/config.js](https://github.com/diegohaz/rest/pull/78/files#diff-d9a3743e726cce2ff7d281b6e4979c25R70)

One thing I added was modifying `Object.assign({}, config.all, config[config.all.env])` to `Object.assign(config.all, config[config.all.env])`, so the `config.all` property would become mutated acquiring the `config[config.all.env]` properties. I usually prefer immutability but this change makes a pretty close to `_` functionality we need.

For now, at least on the code that is generated, it works and `Object.assign` gets its work done correctly, I think that if someone needs specific functionality (the only thing I see that might be a problem is recursive and deep merge) from lodash it can be added.

Let me know if you want me to restore lodash on this PR 😃

DanMMX commented:
Here's the Deep Clone case where I see `_` deserves to stay [Object.assign](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Deep_Clone), not really needed for the generated project since the needed properties are keys from the object.

DanMMX commented:
Any thoughts on this @diegohaz ? 😃

diegohaz commented:
I will proper review this as soon as possible. Thank you so much.
